src/frontend/src/lib/components/networks/MainnetNetwork.svelte

```diff
@@ -0,0 +1,20 @@
+<script lang="ts">
+	import type { Network } from '$lib/types/network';
+	import NetworkComponent from '$lib/components/networks/Network.svelte';
+	import { isNetworkIdEthereum, isNetworkIdICP } from '$lib/utils/network.utils';
+	import {
+		enabledMainnetIcTokensTotalUsd,
+		enabledMainnetErc20TokensTotalUsd
+	} from '$lib/derived/tokens.derived';
+
+	export let network: Network;
+
+	let totalUsd: number | undefined;
```

Comment:
I think you can really make it easier, doing the sum of the store `combinedDerivedSortedNetworkTokensUi` and filtering by networkId (or not filtering for ChainFusion)

Reply:
as discussed, better your solution, to reduce it by one loop